File: leet_2319_x_matrix.go
func checkXMatrix(grid [][]int) bool {
    len := len(grid)
    r := 0
    for r < len {
        c := 0
        for c < len {
            if (r == c || c == len - r - 1) && grid[r][c] == 0 {
                return false
            } else if r != c && c != len - r - 1 && grid[r][c] != 0 {
                return false
            }
            c++
        }
        r++
    }
    return true
}

File: leet_2319_x_matrix.ts
function checkXMatrix(grid: number[][]): boolean {
    let side = grid.length
    let r: number = -1, c: number
    while (++r < side) {
        c = -1
        while (++c < side) {
            if (c == r || c == side - r - 1) {
                if ( !grid[r][c]) {
                    return false
                }
            } else {
                if (grid[r][c]) {
                    return false
                }
            }
        }
    }
    return true
};
